packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -6,18 +6,56 @@ import { FormResourceService } from '../openmrs-api/form-resource.service';
 import { FormSchemaCompiler } from '@openmrs/ngx-formentry';
 import { LocalStorageService } from '../local-storage/local-storage.service';
 import { FormSchema, Questions } from '../types';
+import { TranslateService } from '@ngx-translate/core';
+import { i18n } from 'i18next';
 
+declare global {
+  interface Window {
+    i18next: i18n; // i18n has language in it
+  }
+}
 @Injectable()
 export class FormSchemaService {
   constructor(
     private formsResourceService: FormResourceService,
     private localStorage: LocalStorageService,
     private formSchemaCompiler: FormSchemaCompiler,
+    private translate: TranslateService,
   ) {}
 
+  public getFormTranslationsByUuid(formUuid: string) {
+    const currentLang = (window as Window).i18next?.language.substring(0, 2).toLowerCase() || 'en';
+    this.translate.setDefaultLang(currentLang);
+
+    const formMetadataObject: any = this.formsResourceService.getFormMetaDataByUuid(formUuid).pipe(take(1));
+
+    return this.getFormTranslationsByUuidFromServer(formUuid).subscribe(({ form, referencedComponents }) => {
+      if (form?.translations) {
+        this.translate.setTranslation(form?.language, form?.translations, true);
+      }
+
+      formMetadataObject.translations = form?.translations;
```

Comment:
Is this line necessary? What is this doing?

Reply:
I was trying to add translations to the formMetaData object.